src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ViewComponentTagHelperPass.cs

```diff
@@ -61,24 +61,25 @@ private void GenerateVCTHClass(ClassDeclarationIRNode @class, TagHelperDescripto
             @class.Children.Add(statement);
         }
 
-        private void RewriteCreateNode(
+        private void RewriteTagHelperNode(
             NamespaceDeclarationIRNode @namespace,
             ClassDeclarationIRNode @class,
-            CreateTagHelperIRNode node)
+            TagHelperIRNode node)
         {
-            var originalTypeName = node.TagHelperTypeName;
-
-            var newTypeName = GetVCTHFullName(@namespace, @class, node.Descriptor);
-            for (var i = 0; i < node.Parent.Children.Count; i++)
+            foreach (var descriptor in node.TagHelperTypes.Keys)
```

Comment:
Since we no longer have a node for create, a `Dictionary<TagHelperDescriptor, string>` was needed so that the type name can be overwritten by VCTH pass.